imod/prepare/cleanup.py

```diff
@@ -0,0 +1,219 @@
+) -> dict[str, GridDataArray]:
+    """
+    Clean up drain data, fixes some common mistakes causing ValidationErrors by
+    doing the following:
+
+    - Cells where conductance <= 0 are deactivated.
+    - Cells where concentration < 0 are set to 0.0.
+    - Cells outside active domain (idomain==1) are removed.
+    - Align NoData: If one variable has an inactive cell in one cell, ensure
+      this cell is deactivated for all variables.
+
+    Parameters
+    ----------
+    idomain: xarray.DataArray | xugrid.UgridDataArray
+        MODFLOW 6 model domain. idomain==1 is considered active domain.
+    elevation: xarray.DataArray | xugrid.UgridDataArray
+        Grid with drain elevations
+    conductance: xarray.DataArray | xugrid.UgridDataArray
+        Grid with conductances
+    concentration: xarray.DataArray | xugrid.UgridDataArray, optional
+        Optional grid with concentrations
+
+    Returns
+    -------
+    dict[str, xarray.DataArray | xugrid.UgridDataArray]
+        Dict of cleaned up grids. Has keys: "elevation", "conductance",
+        "concentration".
+    """
+    # Output dict
+    output_dict = {
+        "elevation": elevation,
+        "conductance": conductance,
+        "concentration": concentration,
+    }
+    return _cleanup_robin_boundary(idomain, output_dict)
+
+
+def cleanup_ghb(
+    idomain: GridDataArray,
+    head: GridDataArray,
+    conductance: GridDataArray,
+    concentration: Optional[GridDataArray] = None,
+) -> dict[str, GridDataArray]:
+    """
+    Clean up general head boundary data, fixes some common mistakes causing
+    ValidationErrors by doing the following:
+
+    - Cells where conductance <= 0 are deactivated.
+    - Cells where concentration < 0 are set to 0.0.
+    - Cells outside active domain (idomain==1) are removed.
+    - Align NoData: If one variable has an inactive cell in one cell, ensure
+      this cell is deactivated for all variables.
+
+    Parameters
+    ----------
+    idomain: xarray.DataArray | xugrid.UgridDataArray
+        MODFLOW 6 model domain. idomain==1 is considered active domain.
+    head: xarray.DataArray | xugrid.UgridDataArray
+        Grid with heads
+    conductance: xarray.DataArray | xugrid.UgridDataArray
+        Grid with conductances
+    concentration: xarray.DataArray | xugrid.UgridDataArray, optional
+        Optional grid with concentrations
+
+    Returns
+    -------
+    dict[str, xarray.DataArray | xugrid.UgridDataArray]
+        Dict of cleaned up grids. Has keys: "head", "conductance",
+        "concentration".
+    """
+    # Output dict
+    output_dict = {
+        "head": head,
+        "conductance": conductance,
+        "concentration": concentration,
+    }
+    return _cleanup_robin_boundary(idomain, output_dict)
+
+
+def cleanup_wel(wel_ds: GridDataset):
+    """
+    Clean up wells
+
+    - Removes wells where the screen bottom elevation exceeds screen top.
+    """
+    deactivate = wel_ds["screen_top"] < wel_ds["screen_bottom"]
```

Comment:
General for this file: We are making adjustemesnt to the model. Should we inform the user and/or log this?

Reply:
Good call, I added the ``standard_log_decorator`` to the cleanup utility functions, to log things.